File: 0001-two-sum/0001-two-sum.java

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> dic= new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int diff=target-nums[i];
            if(dic.containsKey(diff))return new int[]{dic.get(diff),i};
            dic.put(nums[i],i);
        }
        return new int[2];
    }
}

File: 0001-two-sum/0001-two-sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $length = count($nums);
        $map = array();
        for($i=0;$i<$length;$i++){
            $complement = $target - $nums[$i];
            if(array_key_exists($complement, $map))
                return array($map[$complement],$i);
            $map[$nums[$i]] = $i;
        }
        return array();
    }

}